File: 3788-MaximumUniqueSubarraySumAfterDeletion/3788-MaximumUniqueSubarraySumAfterDeletion.py
# Last updated: 07/08/2025, 19:30:56
class Solution(object):
    def maxSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        num=[i for i in set(nums) if i>0]
        return max(nums) if len(num)==0 else sum(num)

File: 3788-MaximumUniqueSubarraySumAfterDeletion/3788-MaximumUniqueSubarraySumAfterDeletion.cpp
// Last updated: 07/08/2025, 19:30:49
class Solution {
public:
    int maxSum(vector<int>& nums) {
        unordered_set<int> positivenums;
        for(int i:nums)
            if(i>0) positivenums.emplace(i);
        if(positivenums.empty())
            return *max_element(nums.begin(),nums.end());
        else
            return accumulate(positivenums.begin(),positivenums.end(),0);
    }
};